packages/library/src/commonMain/kotlin/io/realm/internal/RealmUtils.kt

```diff
@@ -105,6 +108,31 @@ fun <T : RealmObject> create(
     }
 }
 
+fun <T> copyToRealmIfNeeded(
```

Comment:
Shouldn't this just not just be collapsed with `copyToRealm`? I think the validity check could might as well be in `copyToRealm` and `copyToRealm` already returns the same element if it is already managed. So looks like you should just also check if it actually is an `RealmObjectInternal` and otherwise return the element. Then we would have multiple methods with almost same functionality.